test/extended/olm/olmv1.go

```diff
@@ -417,3 +622,92 @@ func checkFeatureCapability(oc *exutil.CLI) {
 		g.Skip("Test only runs with OperatorLifecycleManagerV1 capability")
 	}
 }
+
+// verifyAPIEndpoint runs a job to validate the given service endpoint of a ClusterCatalog
+func verifyAPIEndpoint(ctx g.SpecContext, oc *exutil.CLI, serviceURL string) {
+	jobName := fmt.Sprintf("test-catalog-endpoint-%s", rand.String(5))
+
+	tempFile, err := os.CreateTemp("", "api-test-job-*.yaml")
+	o.Expect(err).NotTo(o.HaveOccurred())
+	defer os.Remove(tempFile.Name())
+
+	jobYAML := fmt.Sprintf(`
```

Comment:
You couldn't just `Fprintf()` directly into the file?

Reply:
+1, but there is also os.WriteFile() that can write `[]byte` directly to a file, no need to deal with open file handles and closing them correctly.